src/protocols/user_directed_commissioning/UserDirectedCommissioningClient.cpp

```diff
@@ -44,19 +44,31 @@ CHIP_ERROR UserDirectedCommissioningClient::SendUDCMessage(TransportMgrBase * tr
     ChipLogProgress(Inet, "Sending UDC msg");
 
     UDCClientContext * udcClientContext = new UDCClientContext();
-
     udcClientContext->mUdcAttemptsRemaining =
         CHIP_DEVICE_CONFIG_TOTAL_UDC_MSG_COUNT; // repeat sending UDC message per spec (no ACK on this message)
     udcClientContext->mTransportMgr      = transportMgr;
     udcClientContext->mTargetPayload     = payload.CloneData();
     udcClientContext->mTargetPeerAddress = peerAddress;
-    DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(0), SendUDCTask, udcClientContext);
+
+    ReturnErrorOnFailure(
+        DeviceLayer::PlatformMgrImpl().AddEventHandler(DeviceEventCallback, reinterpret_cast<intptr_t>(udcClientContext)));
```

Comment:
Can you please move this to the class constructor and then add removeEventHandler in the destructor? I'm not sure what behavior would we see if we invoke the function multiple times. Would the callback be triggered n times?

Reply:
Doing that crashes the app with "CHIP: [SPT] ABORT: chip::Platform::MemoryAlloc() called before chip::Platform::MemoryInit()" because the constructor executes too soon i.e. before MemoryInit(). Thinking of alternatives...